### PR TITLE
Tool for debugging ring-buffer-crash.dump

### PR DESCRIPTION
**Pull request by joestringer**
The dumps from the bpf package are difficult to parse by eye or hand.
This tool takes such a file in as input and attempts to render it into
something readable by humans.

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/5873)

### CODE REVIEW

**Comment by joestringer** [comment]
test-me-please

**Comment by joestringer** [comment]
test-me-please

EDIT: imports ordering style breakage. Fixed.

**Comment by joestringer** [comment]
test-me-please

**Comment by joestringer** [comment]
test-me-please

File: pkg/bpf/perf.go [hunk 1]
```diff
@@ -652,3 +671,97 @@ func (e *PerCpuEvents) CloseAll() error {
 
 	return retErr
 }
+
+// decode uses reflection to read bytes directly from 'reader' into the object
+// pointed to from 'i'. Assumes that 'i' is a pointer.
+func decode(i interface{}, reader io.ReadSeeker) error {
```
Comment (aanm) [inline pkg/bpf/perf.go:679]: Doesn't reflection make everything slow?

**Review by aanm — APPROVED** [review]

File: pkg/bpf/perf.go (hunk 1 above)
Comment (joestringer) [reply pkg/bpf/perf.go:679]: For an single loop out-of-order across one ringbuffer in a tool that is run directly by a user, it's fine. I can add a comment that this function and it's users should not be used from perfomance-sensitive paths.

**Review by ianvernon — APPROVED** [review]

**Comment by joestringer** [comment]
Green CI run, will add comments to the `Decode()` functions to describe their intent wrt performance.

File: pkg/bpf/perf.go (hunk 1 above)
Comment (joestringer) [reply pkg/bpf/perf.go:679]: Done.

**Comment by joestringer** [comment]
test-me-please

**Review by tgraf — APPROVED** [review]

**Pull request merged**